### DISCUSSION TITLE
Digitalocean droplet setup

### DISCUSSION THREAD

**Discussion by toddabraham** [smartcontractkit/full-blockchain-solidity-course-py#24] [Q&A] [answered]
Hey there everyone.  I'm really enjoying this course, it's opening up my understanding of blockchain in ways I never imagined.

I'm running on an old (2015ish) Macbook Air and trying to offload as much work from local resources as possible.  I've setup a Docker / Ubuntu container on Digitalocean and trying to SSH in and edit files on the server using VSCode.  I just ran the deply.py file on the server and everything seems to be working ok.  The does seem to be an issue with the Solidity extension though, it keeps crashing the SSH connection.  I've uninstalled Solidity and the connection stabilizes.  Any thoughts on what maybe happening?

Also, do I need to install a solidity compiler to my DO droplet?  I haven't done that, could that be what's causing the issue?  

Thanks in advance to the group for any guidance on this.

**Comment by toddabraham** [comment]
UPDATE: The connection seems to have stabilized.  I'm not sure if installing py-solc-x did it or, but it's mostly working.

NEW PROBLEM: I'm at the stage of saving and running my deploy.py file, and now running into trouble.  Issuing terminal command, sudo python3 deploy.py returns SolcNotInstalled(solcx.exceptions.SolcNotInstalled: solc 0.6.0 has not been installed.  Use solcx.install_solc('0.6.0') to install.

Tried to follow the solcx install syntax and had bash: syntax error near unexpected token '('

![image](https://user-images.githubusercontent.com/26919212/133579788-b16add22-01cf-4b2d-8b4c-092699d6e2d3.png)

**Reply by AlbertoSinigaglia** [reply]
sorry, you are trying to install solcx using what? have you checked out [this issue](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/issues/21)?

**Reply by toddabraham** [reply]
With high hopes, I ran through the issue link you shared.  I had a venv already setup, so thought maybe running pip3 install py-solc-x would resolve, but sadly no.  python deploy.py returns the same.

**Comment by PatrickAlphaC** [comment, marked as answer]
Or have you seent this: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/5

**Reply by toddabraham** [reply]
Mucho gracias señor, that worked perfectly!  I was confused by the above instructions and kept trying to run import solcx and solcx.install....at the command line.

**Comment by toddabraham** [comment]
But....the SSH disconnection issue remains.  It seems as long as I'm working in my deploy.py file, no issues occur, the connection is stable.

However, as soon as I touch my .sol file, the issue occurs....

![image](https://user-images.githubusercontent.com/26919212/133651283-8ff26f27-36a5-4478-96b5-3df058676297.png)

**Reply by PatrickAlphaC** [reply]
Anything here help?

https://github.com/microsoft/vscode-remote-release/issues/4686

**Comment by toddabraham** [comment]
EDIT: Fixed the issue by rebuilding my system. Stabilized now.

Seems to have stabilized a bit, not dropping as often at least.

**Reply by toddabraham** [reply]
Spoke too soon!  Into the Brownie setup and when creating the new SimpleStorage.sol file in brownie/contracts the issue continues, to the point where I can't save the file to the server.  Not exactly a true statement, it does save, but it takes many attempts & SSH reconnects.

**Reply by toddabraham** [reply]
Fixed the issue by rebuilding my system.  Stabilized now.
